lib/flv/transmuxer.js

```diff
@@ -227,26 +227,35 @@ VideoSegmentStream = function(track) {
       return;
     }
 
+    // Throw away nalUnits at the start of the byte stream until we find
+    // the first AUD
+    while (nalUnits.length) {
+      if (nalUnits[0].nalUnitType === 'access_unit_delimiter_rbsp') {
+        break;
+      }
+      nalUnits.shift();
+    }
+
     while (nalUnits.length) {
       currentNal = nalUnits.shift();
 
-    // record the track config
-    if (currentNal.nalUnitType === 'seq_parameter_set_rbsp') {
-      track.newMetadata = true;
-      config = currentNal.config;
-      track.width = config.width;
-      track.height = config.height;
-      track.sps = [currentNal.data];
-      track.profileIdc = config.profileIdc;
-      track.levelIdc = config.levelIdc;
-      track.profileCompatibility = config.profileCompatibility;
-      h264Frame.endNalUnit();
-    } else if (currentNal.nalUnitType === 'pic_parameter_set_rbsp') {
-      track.newMetadata = true;
-      pps = currentNal.data;
-      track.pps = [currentNal.data];
-      h264Frame.endNalUnit();
-    } else if (currentNal.nalUnitType === 'access_unit_delimiter_rbsp') {
+      // record the track config
+      if (currentNal.nalUnitType === 'seq_parameter_set_rbsp') {
+        track.newMetadata = true;
+        config = currentNal.config;
+        track.width = config.width;
+        track.height = config.height;
+        track.sps = [currentNal.data];
+        track.profileIdc = config.profileIdc;
+        track.levelIdc = config.levelIdc;
+        track.profileCompatibility = config.profileCompatibility;
+        h264Frame.endNalUnit();
+      } else if (currentNal.nalUnitType === 'pic_parameter_set_rbsp') {
+        track.newMetadata = true;
```

Comment:
Tangential to this review but why is this boolean necessary? Wouldn't `track.pps.length` suffice?